src/main.js

```diff
@@ -74,14 +75,45 @@ let load = loader({
 
   github: {
     requires: ['cfg'],
-    setup: ({cfg}) => {
+    setup: async ({cfg}) => {
       let github = new Github({
         promise: Promise,
       });
-      if (cfg.github.credentials.token) {
-        github.authenticate(cfg.github.credentials);
-      }
-      return github;
+
+      // This object insures that the authentication is delayed until we need it.
+      // Also, the authentication happens not just once in the beginning, but for each request.
+      return {
+        getInstallationGithub: async (inst_id) => {
+          let inteToken = jwt.sign(
+            {
+              iss: cfg.github.credentials.integrationId,
+            },
+            cfg.github.credentials.privatePEM,
+            {
+              algorithm: 'RS256',
+              expiresIn: '1m',
+            }
+          );
+          debug('Authenticating as integration...');
```

Comment:
This process will happen a lot, so let's limit it to one debug message per call to `getInstallationGithub` -- probably just the last one (for success)